source/intro.tex

```diff
@@ -792,23 +792,37 @@
 \end{footnote}
 that program.
 \item
+\indextext{behavior!undefined}%
+If a program contains a violation of a rule for which no diagnostic is required,
+this document places no requirement on implementations
+with respect to that program.
+\item
 \indextext{message!diagnostic}%
-If a program contains a violation of any diagnosable rule or an occurrence
+Otherwise, if a program contains a violation of any diagnosable rule or an occurrence
 of a construct described in this document as ``conditionally-supported'' when
 the implementation does not support that construct, a conforming implementation
 shall issue at least one diagnostic message.
-\item
-\indextext{behavior!undefined}%
-If a program contains a violation of a rule for which no diagnostic
-is required, this document places no requirement on
-implementations with respect to that program.
 \end{itemize}
 \begin{note}
 During template argument deduction and substitution,
 certain constructs that in other contexts require a diagnostic
 are treated differently;
 see~\ref{temp.deduct}.
 \end{note}
+Furthermore, a conforming implementation
+\begin{itemize}
+\item
+shall not accept a preprocessing translation unit containing
+a \tcode{\# error} preprocessing directive\iref{cpp.error},
+\item
+shall issue at least one diagnostic message for
+each \tcode{\# warning} or \tcode{\# error} preprocessing directive
+not following a \tcode{\# error} preprocessing directive in
```

Comment:
```suggestion
a \tcode{\#error} preprocessing directive\iref{cpp.error},
\item
shall issue at least one diagnostic message for
each \tcode{\#warning} or \tcode{\#error} preprocessing directive
not following a \tcode{\#error} preprocessing directive in
```
Looking at `preprocessor.tex`, a space is only used in a `ncsimplebnf` environment.

Reply:
Agreed. Please remove those spaces between # and the directive name.